Mathlib/Data/Matrix/Basis.lean

```diff
@@ -255,8 +257,12 @@ theorem mem_range_scalar_of_commute_stdBasisMatrix {M : Matrix n n α}
     · rw [col_eq_zero_of_commute_stdBasisMatrix (hM k i hkl.symm) hkl]
     · rw [row_eq_zero_of_commute_stdBasisMatrix (hM i j hij) hkl.symm]
 
-end Commute
+theorem mem_range_scalar_iff_commute_stdBasisMatrix {M : Matrix n n α} :
+    M ∈ Set.range (Matrix.scalar n) ↔ ∀ (i j : n), i ≠ j → Commute (stdBasisMatrix i j 1) M := by
```

Comment:
Seems like it might be useful to have a version without the condition `i ≠ j` for convenience.

Reply:
For the forward direction you can replace `stdBasisMatrix i j 1` by any square matrix :)

Reply:
Surely only if `α`is commutative? Or are we already in that case?

Reply:
Apparently not; my bad!

Reply:
Now we have `mem_range_scalar_iff_commute_stdBasisMatrix'` for this purpose.